rqt_gui_cpp/src/rqt_gui_cpp/roscpp_plugin_provider.cpp

```diff
@@ -30,52 +30,56 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "roscpp_plugin_provider.h"
+#include "roscpp_plugin_provider.hpp"  // NOLINT
```

Comment:
Same here; does this really need to be `NOLINT`?

Reply:
I'm getting a wierd error

```
src/rqt_gui_cpp/nodelet_plugin_provider.cpp:33: rqt_gui_cpp/src/rqt_gui_cpp/nodelet_plugin_provider.cpp should include its header file rqt_gui_cpp/src/rqt_gui_cpp/nodelet_plugin_provider.h
```

Reply:
Also this check is spurious

Reply:
Actually, I just realized why this is weird.  That is because both `nodelet_plugin_provider.hpp` and `roscpp_plugin_provider.hpp` are *private* headers.  Thus, we don't need to deprecate them at all, we can just rename them to have the `.hpp` extension, and all should be good with them.  So I'm going to suggest we do that, which should let us get rid of the NOLINT here and above.